File: A09 two pointer(sliding Window)/002 two pointers/160. Intersection of Two Linked Lists/Leetcode.java
/*
Time Complexity: O(N)
Space Complexity: O(1)

*/ 


public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode a = headA, b = headB;
        while (a != b) {
            a = a == null ? headB : a.next;  
	    // if (a == null) a = headB;
	//	else a = a.next


            b = b == null ? headA : b.next;
        }
        return a;
    }
}

File: A06 Nodes Related/004 LinkedList/Remove dupliated Sorted List/Leetcode 82 Remove Duplicated From Sorted Lists II/Leetcode.py
###########
Time Complexity:
Space Complexity:


#############
class Solution:
    def deleteDuplicates(self, head):
        p = head
        q = None
        while p and p.next: 
            if p.val == p.next.val: 
                while (p and p.next and p.val == p.next.val): 
                    p = p.next
                p = p.next
                if q is None:
                    head = p
                else:
                    q.next = p
            else:
                q = p
                p = p.next
        return head